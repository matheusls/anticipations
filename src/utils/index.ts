export const centsToReal = (cents: number) => {
  return new Intl.NumberFormat('pt-BR', {
    style: 'currency',
    currency: 'BRL',
  }).format(cents / 100);
};

export const realToCents = (real: string) => {
  return Number(real.replace(/\D+/g, ''));
};
