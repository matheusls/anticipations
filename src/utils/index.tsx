import { toast, TypeOptions } from 'react-toastify';

export const centsToReal = (cents: number) => {
  return new Intl.NumberFormat('pt-BR', {
    style: 'currency',
    currency: 'BRL',
  }).format(cents / 100);
};

export const realToCents = (real: string) => {
  return Number(real.replace(/\D+/g, ''));
};

type ErrorToastProps = {
  text: string;
  type: TypeOptions;
};

export const renderErrorToast = (error = '') => {
  const errorToastProps: ErrorToastProps = {
    '500': {
      text: 'Oops, tivemos um problema. Por favor, tente novamente.',
      type: 'error' as const,
    },
    '408': {
      text: 'Oops, nossos servidores estão demorando para responder...',
      type: 'warning' as const,
    },
    offline: {
      text: 'Você parece estar com problemas de conexão!',
      type: 'warning' as const,
    },
  }[error] || {
    text: 'Oops, houve um problema. Por favor, tente novamente.',
    type: 'error' as const,
  };

  const { text, type } = errorToastProps;

  return toast(text, {
    position: 'top-center',
    autoClose: false,
    closeOnClick: true,
    type,
  });
};
