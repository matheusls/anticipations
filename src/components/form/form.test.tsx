import { render, screen } from 'tests/utils';

import Form from './form';

describe('<Form />', () => {
  it('should match snapshot', () => {
    const { container } = render(<Form>Form component</Form>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render Form correctly', () => {
    render(<Form aria-label="form">Form component</Form>);

    expect(screen.getByRole('form')).toBeInTheDocument();
  });
});
