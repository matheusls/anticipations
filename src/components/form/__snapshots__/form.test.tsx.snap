// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should match snapshot 1`] = `
.c0 {
  background-color: #fff;
  margin: 0;
  padding: 41px 56px;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

<form
  class="c0"
>
  Form component
</form>
`;
